### DISCUSSION TITLE
How to get no-use-before-define for in-scope variables only?

### DISCUSSION THREAD

**Discussion by ghost** [eslint/eslint#16518] [Q&A] [answered]
Hello, when I enable `no-use-before-define`, this functioning code gets flagged:

```js
export const SomeThing  = {
  doStuff() {
    console.log("TEST", xyz);
  },
};

const xyz = "XYZ";
```

Is there a way that I can disable this behavior while still getting warned about bad code such as the following?

```js
export const SomeThing  = {
  doStuffToo() {
    console.log("TEST", abc);
    const abc = "ABC";
  },
};
```

**Comment by mdjermanovic** [comment, marked as answer]
You can set [`"variables": false`](https://eslint.org/docs/latest/rules/no-use-before-define#variables) option to allow references to variables in upper scopes.

[Playground demo link](https://eslint.org/play/#eyJ0ZXh0IjoiZXhwb3J0IGNvbnN0IFNvbWVUaGluZyAgPSB7XG4gIGRvU3R1ZmYoKSB7XG4gICAgY29uc29sZS5sb2coXCJURVNUXCIsIHh5eik7IC8vIG5vIGVycm9yXG4gIH0sXG59O1xuXG5jb25zdCB4eXogPSBcIlhZWlwiO1xuXG5leHBvcnQgY29uc3QgU29tZVRoaW5nMiAgPSB7XG4gIGRvU3R1ZmZUb28oKSB7XG4gICAgY29uc29sZS5sb2coXCJURVNUXCIsIGFiYyk7IC8vIGVycm9yXG4gICAgY29uc3QgYWJjID0gXCJBQkNcIjtcbiAgfSxcbn07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby11c2UtYmVmb3JlLWRlZmluZSI6WyJlcnJvciIseyJ2YXJpYWJsZXMiOmZhbHNlfV19LCJlbnYiOnsiZXM2Ijp0cnVlfX19)

**Reply by ghost** [reply]
Thanks so much for that demo. I had previously tried that configuration in my own code and it didn’t work, so now I know that there must be some other issue with my configuration perhaps due to a mixed type script and JavaScript project.
